pandas/core/reshape/merge.py

```diff
@@ -1263,8 +1263,8 @@ def _get_merge_keys(self):
             else:
                 lt = left_join_keys[-1]
 
-            msg = "incompatible tolerance, must be compat " \
-                  "with type {0}".format(type(lt))
+            msg = "incompatible tolerance {0}, must be compat " \
+                  "with type {1}".format(type(tolerance), lt.dtype)
```

Comment:
1) Let's use keyword arguments instead of `{0}` and `{1}`.

2) Is there a test to check this error message?  If not, let's add one.